controllers/validate.go

```diff
@@ -0,0 +1,83 @@
+// Package controllers is the layer for imput date in the application.
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// DockerRequestPayloadNew is sent by docker hub whenever a new push happen to a
+// repository.
+type DockerRequestPayloadNew struct {
+	CallbackURL string `json:"callback_url"`
+	PushData    struct {
+		Images   []string `json:"images"`
+		PushedAt int      `json:"pushed_at"`
+		Pusher   string   `json:"pusher"`
+		Tag      string   `json:"tag"`
+	} `json:"push_data"`
+	Repository struct {
+		CommentCount    int    `json:"comment_count"`
+		DateCreated     int    `json:"date_created"`
+		Description     string `json:"description"`
+		Dockerfile      string `json:"dockerfile"`
+		FullDescription string `json:"full_description"`
+		IsOfficial      bool   `json:"is_official"`
+		IsPrivate       bool   `json:"is_private"`
+		IsTrusted       bool   `json:"is_trusted"`
+		Name            string `json:"name"`
+		Namespace       string `json:"namespace"`
+		Owner           string `json:"owner"`
+		RepoName        string `json:"repo_name"`
+		RepoURL         string `json:"repo_url"`
+		StarCount       int    `json:"star_count"`
+		Status          string `json:"status"`
+	} `json:"repository"`
+}
+	
+// DockerValidate is POST by us for validate the process.
+// The following parameters are recognized in callback data:
+// state (required): Accepted values are success, failure, and error. If the state isn’t success, the Webhook chain is interrupted.
+// description: A string containing miscellaneous information that is available on Docker Hub. Maximum 255 characters.
+// context: A string containing the context of the operation. Can be retrieved from the Docker Hub. Maximum 100 characters.
+// target_url: The URL where the results of the operation can be found. Can be retrieved on the Docker Hub.
```

Comment:
Please do not cross the 100 columns barrier, please break this comments into multiple lines.